language/vm/vm_runtime/src/execution_stack.rs

```diff
@@ -40,7 +41,11 @@ where
     pub fn push_call(&mut self, function: FunctionRef<'txn>) -> VMResult<()> {
         let callee_arg_size = function.arg_count();
         let args = self.popn(callee_arg_size as u16)?;
-        self.function_stack.push(Frame::new(function, args));
+        if self.function_stack.len() < usize::max_value() {
+            self.function_stack.push(Frame::new(function, args));
```

Comment:
Nit: could move the`Ok` from 49 into this branch to make the code a bit more symmetrical/avoid `return`s (and similar below for `push_frame`).